src/arvgvfakecamera.c

```diff
@@ -483,20 +500,46 @@ arv_gv_fake_camera_start (ArvGvFakeCamera *gv_fake_camera)
 			}
 			g_clear_object (&inet_socket_address);
 			g_clear_object (&inet_address);
-
-			g_free (gvcp_address_string);
+                        
+                        /* subnet broadcast discovery */
+			socket_address = g_socket_address_new_from_native (ifap_iter->ifa_broadaddr,
+									   sizeof (struct sockaddr));
+                        inet_address = g_inet_socket_address_get_address (G_INET_SOCKET_ADDRESS (socket_address));
+			discovery_address_string = g_inet_address_to_string (inet_address);
+			arv_debug_device ("[GvFakeCamera::start] Discovery address = %s", discovery_address_string);
 			g_free (discovery_address_string);
+			inet_socket_address = g_inet_socket_address_new (inet_address, ARV_GVCP_PORT);
+			if (g_strcmp0 (gvcp_address_string, discovery_address_string) == 0)
+				gv_fake_camera->priv->subnet_discovery_socket = NULL;
+			else {
+				gv_fake_camera->priv->subnet_discovery_socket = g_socket_new (G_SOCKET_FAMILY_IPV4,
+									    G_SOCKET_TYPE_DATAGRAM,
+									    G_SOCKET_PROTOCOL_UDP, NULL);
+				if (!g_socket_bind (gv_fake_camera->priv->subnet_discovery_socket, inet_socket_address, FALSE, NULL))
+					binding_error = TRUE;
+				g_socket_set_blocking (gv_fake_camera->priv->subnet_discovery_socket, FALSE);
+			}
+			g_clear_object (&socket_address);
+			g_clear_object (&inet_socket_address);
+			g_clear_object (&inet_address);
```

Comment:
You don't own inet_address here

Reply:
Right - this is exactly why I hate GTK. You have to consult the document to know whether an object's ownership is passed or not (providing you can rely on it), rather than being able to rely on the compiler to enforce ownership semantics for you.

Anyway, hopefully fixed now.